sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/WindowsCredentialApi.java

```diff
@@ -0,0 +1,294 @@
+         *    This member cannot be longer than CRED_MAX_STRING_LENGTH (256) characters.
+         */
+        public String Keyword;
+
+        /**
+         *   Identifies characteristics of the credential attribute. This member is reserved and should be originally
+         *   initialized as zero and not otherwise altered to permit future enhancement.
+         */
+        public int Flags;
+
+        /**
+         *   Length of Value in bytes. This member cannot be larger than CRED_MAX_VALUE_SIZE (256).
+         */
+        public int ValueSize;
+
+        /**
+         *   Data associated with the attribute. By convention, if Value is a text string, then Value should not
+         *   include the trailing zero character and should be in UNICODE.
+         *
+         *   Credentials are expected to be portable. The application should take care to ensure that the data in
+         *   value is portable. It is the responsibility of the application to define the byte-endian and alignment
+         *   of the data in Value.
+         */
+        public Pointer Value;
+    }
+
+
+    /**
+     * The CREDENTIAL structure contains an individual credential
+     *
+     * typedef struct _CREDENTIAL {
+     *   DWORD                 Flags;
+     *   DWORD                 Type;
+     *   LPTSTR                TargetName;
+     *   LPTSTR                Comment;
+     *   FILETIME              LastWritten;
+     *   DWORD                 CredentialBlobSize;
+     *   LPBYTE                CredentialBlob;
+     *   DWORD                 Persist;
+     *   DWORD                 AttributeCount;
+     *   PCREDENTIAL_ATTRIBUTE Attributes;
+     *   LPTSTR                TargetAlias;
+     *   LPTSTR                UserName;
+     * } CREDENTIAL, *PCREDENTIAL;
+     */
+    class CREDENTIAL extends Structure {
+
+        @Override
+        protected List<String> getFieldOrder() {
+            return Arrays.asList("Flags",
+                    "Type",
+                    "TargetName",
+                    "Comment",
+                    "LastWritten",
+                    "CredentialBlobSize",
+                    "CredentialBlob",
+                    "Persist",
+                    "AttributeCount",
+                    "Attributes",
+                    "TargetAlias",
+                    "UserName");
+        }
+
+        public CREDENTIAL() {
+            super();
+        }
+
+        public CREDENTIAL(final int size) {
+            super(new Memory(size));
+        }
+
+        public CREDENTIAL(Pointer memory) {
+            super(memory);
+            read();
+        }
+
+        /**
+         *   A bit member that identifies characteristics of the credential. Undefined bits should be initialized
+         *   as zero and not otherwise altered to permit future enhancement.
+         *
+         *   See MSDN doc for all possible flags
+         */
+        public int Flags;
```

Comment:
variable names should start from lower case.

Reply:
maps to native C structs fields.